fdbclient/TaskBucket.h

```diff
@@ -195,14 +195,20 @@ class TaskBucket : public ReferenceCounted<TaskBucket> {
 		return runRYWTransaction(cx, [=](Reference<ReadYourWritesTransaction> tr) { return getOne(tr); });
 	}
 
-	Future<bool> doTask(Database cx, Reference<FutureBucket> futureBucket, Reference<Task> task);
+	Future<bool> doTask(Database cx,
+	                    Reference<FutureBucket> futureBucket,
+	                    Reference<Task> task,
+	                    Optional<std::string> proxy);
```

Comment:
Adding an parameter for `TaskBucket` is awkward, because it's a setting unrelated to the abstraction.

A better (still not ideal) way may be just declare a global variable for `FileBackupAgent` to use, which can be set by the backup agent.

Reply:
thanks, yeah this PR is just for demo purpose and is not ready yet, but the advise makes sense